src/azure-cli/azure/cli/command_modules/iot/custom.py

```diff
@@ -499,6 +510,15 @@ def update_iot_hub_custom(instance,
     if fileupload_notification_ttl is not None:
         ttl = timedelta(hours=fileupload_notification_ttl)
         instance.properties.messaging_endpoints['fileNotifications'].ttl_as_iso8601 = ttl
+
+    identityBased_fileUpload = fileupload_storage_authentication_type and fileupload_storage_authentication_type.lower() == AuthenticationType.IdentityBased.value
+    if identityBased_fileUpload:
+        instance.properties.storage_endpoints['$default'].authentication_type = AuthenticationType.IdentityBased
+        instance.properties.storage_endpoints['$default'].container_uri = fileupload_storage_container_uri
+    else:
```

Comment:
We should check whether the fileupload_storage_authentication_type is None or not. User just doesn't want to change anything. According to current logic, the container_uri would be set to None instead.